model-optimizer/mo/utils/broadcasting.py

```diff
@@ -79,6 +79,25 @@ def bi_directional_shape_broadcasting(input_shape_1: np.array, input_shape_2: np
     return np.maximum(shape_1, shape_2)
 
 
+def explicit_shape_broadcasting(input_shape: np.array, target_shape: np.array, axes_mapping: np.array) -> np.array:
+    """
+    Explicit shape broadcasting of input tensor. Function only asserts that values are correct.
+    Resulting shape is equal to target_shape.
+    :param input_value: input value to broadcast
+    :param target_shape: target shape
+    :param axes_mapping: a list of axis indices, each index maps an axis from the input_value to axis in the output
+    :return: broadcasted shape
+    """
+    assert np.all(np.diff(axes_mapping) >= 0), "axes_mapping is not sorted"
+    assert len(axes_mapping) == len(input_shape), "size of axes_mapping does not match to rank of input"
+
+    res = target_shape.copy()
+    for i, axis in enumerate(axes_mapping):
+        assert axis < len(res), "axis value from axes_mapping exceeds rank of target_shape"
```

Comment:
do we have such a requirement -- I haven't found one here: https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/movement/Broadcast_3.md
Neither I found axes normalization in the code. Did you talked it through with @lazarevevgeny?

Reply:
>do we have such a requirement -- I haven't found one here:...

This restriction is not described in specification but it comes from "common sense". E.g. lets assume that `target_shape = (16, 3, 100, 100)` then `axes_mapping = [5]` is meaningless.

About other questions, I think it will be better to discuss personally.

Reply:
Added axes normalization and PermuteInputs for axes_mapping. Please review.

Reply:
shouldn't we have it here too? not only for value inferring?

Reply:
Agreed will all comment and corrected asserts and normalizations. Please review